src/navier_stokes/INSStaggeredHierarchyIntegrator.cpp

```diff
@@ -1440,15 +1490,28 @@ INSStaggeredHierarchyIntegrator::setupSolverVectors(const Pointer<SAMRAIVectorRe
             d_convective_op->setSolutionTime(apply_time);
             d_convective_op->apply(*d_U_adv_vec, *d_N_vec);
         }
+
         const int N_idx = d_N_vec->getComponentDescriptorIndex(0);
         if (convective_time_stepping_type == ADAMS_BASHFORTH)
         {
 #if !defined(NDEBUG)
             TBOX_ASSERT(cycle_num == 0);
 #endif
-            const double omega = dt / d_dt_previous[0];
-            d_hier_sc_data_ops->linearSum(N_idx, 1.0 + 0.5 * omega, N_idx, -0.5 * omega, d_N_old_current_idx);
+            const double omega = getTimeStepSizeRatio();
+            double beta1, beta2;
+            if (is_bdf_time_stepping_type(d_viscous_time_stepping_type))
+            {
+                beta1 = 1.0 + omega;
+                beta2 = -omega;
+            }
```

Comment:
I don't think this is true for `BACKWARD_EULER`. The coefficients depend on the particular BDF scheme.

Reply:
Don't you want the RHS to be at the end of the time step at least for any BDF with order >= 2?

Reply:
Yes, but `BACKWARD_EULER` is not order >= 2. I guess if you are using first order viscous time stepping, using first order convective time stepping is fine too?

Reply:
Well, what is done now is to use time-step centered convective time stepping with `BACKWARD_EULER`.

I can drop `BACKWARD_EULER` from the list of BDF time stepping schemes if you would prefer that. 😁

Reply:
I have removed `BACKWARD_EULER` from the list of BDF time stepping schemes for now. Maybe at some point we can rationalize this stuff.